### DISCUSSION TITLE
After modifying the compactor_blocks_retention_period parameter, do you operate on historical blocks?

### DISCUSSION THREAD

**Discussion by hexiang1219** [grafana/mimir#6104] [Help and support] [answered]
Assume that the Compactor_blocks_retention_period parameter is configured to 30d. After two months, the parameter changes to 7d. So after this parameter takes effect, will blocks other than 7d and within 30d be deleted?

**Comment by pstibrany** [comment, marked as answer]
> So after this parameter takes effect, will blocks other than 7d and within 30d be deleted?

Yes, on its next run, compactor's block cleanup process will use current vlaue of "block retention period" to decide whether to delete the block or not.

**Comment by hexiang1219** [comment]
OK Thanks
